src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/AmbientValueAttribute.cs

```diff
@@ -22,6 +24,13 @@ public AmbientValueAttribute([DynamicallyAccessedMembers(DynamicallyAccessedMemb
         {
             // The try/catch here is because attributes should never throw exceptions. We would fail to
             // load an otherwise normal class.
+
+            if (!IDesignerHost.IsSupported)
+            {
+                Debug.Assert(!IDesignerHost.IsSupported, "Runtime instantiation of this attribute is not allowed.");
```

Comment:
Just curious - it feels weird asserting that the condition of the if is true in the true branch...

Reply:
The assert is meant to give an indication of the problem of using this attribute when the feature switch is set. Throwing (with an appropriate message) here would be ideal but it seems prudent to respect the comments on the type. Consumers (like WinForms) could also generate a warning via its code generator when it detects use of this attribute at runtime if the feature switch is set.